File: src/main/java/fr/openent/crre/model/WorkflowNeo4jModel.java
package fr.openent.crre.model;

import fr.openent.crre.helpers.IModelHelper;
import io.vertx.core.json.JsonObject;
import fr.openent.crre.core.constants.Field;


public class WorkflowNeo4jModel implements IModel<WorkflowNeo4jModel> {
    private String displayName;
    private String name;

    public WorkflowNeo4jModel() {
    }

    public WorkflowNeo4jModel(JsonObject jsonObject) {
        this.displayName = jsonObject.getString(Field.DISPLAYNAME);
        this.name = jsonObject.getString(Field.NAME);
    }

    @Override
    public JsonObject toJson() {
        return IModelHelper.toJson(this, true, true);
    }

    public String getDisplayName() {
        return displayName;
    }

    public WorkflowNeo4jModel setDisplayName(String displayName) {
        this.displayName = displayName;
        return this;
    }

    public String getName() {
        return name;
    }

    public WorkflowNeo4jModel setName(String name) {
        this.name = name;
        return this;
    }
}

File: src/main/resources/public/ts/model/WorkflowNeo4jModel.ts
export interface IWorkflowNeo4jModel {
    name: string
    display_name: string
}

export class WorkflowNeo4jModel implements IWorkflowNeo4jModel{
    private _name: string
    private _display_name: string

    constructor(data: IWorkflowNeo4jModel) {
        this._name = data.name;
        this._display_name = data.display_name;
    }

    get name(): string {
        return this._name;
    }
    set name(value: string) {
        this._name = value;
    }


    get display_name(): string {
        return this._display_name;
    }

    set display_name(value: string) {
        this._display_name = value;
    }
}